Automattic/newspack-network
# Epic/network memberships handling

This epic branch adds handling of network subscriptions and memberships based on the data made available by https://github.com/Automattic/newspack-network/pull/122

It should be merged only once https://github.com/Automattic/newspack-network/pull/122 has been deployed and we have executed the backfills on every site.

The first PR in this epic branch reverts https://github.com/Automattic/newspack-network/pull/114. (merged)

## PRs to be merged in order

###  1: https://github.com/Automattic/newspack-network/pull/135

This is a re-implementation of https://github.com/Automattic/newspack-network/pull/114

Testing:

* Create a granted-on-subscription memberships plan on two network sites, and make them synchronised (set same Network ID in the plan settings)
* On the Hub, buy a subscription tied to that membership plan as a reader
* On the Node, Pull all events
* Visit the Node as an anonymous reader and buy a subscription that gives access to the same membership plan using the same email
* Run  `wp newspack-network process-webhooks` in the node to sync events to the Hub
* On the Hub admin, Cancel the subscription
* Confirm the associated Membership is kept active, and that a note is added to the user membership saying it was kept active because there is an active subscription in the Node

###  4: https://github.com/Automattic/newspack-network/pull/137

This is a re-implementation of https://github.com/Automattic/newspack-network/pull/112

Testing: 

* Create a granted-on-subscription memberships plan on two network sites, and make them synchronised (set same Network ID in the plan settings)
* Make sure to sync the latest events by running  `wp newspack-network process-webhooks` and `wp newspack-network sync-all` in the Node 
* you should see the `newspack_network_membership_plan_updated` events in the Event log for both the plan in the hub and the plan in the node
* you also should see that the `newspack_network_membership_plans` option populated in both sites (this is what we tested in #124 
* One one site, buy the subscription, which should grant you the plan
* On the second site, log in/register using the same email address
* On the second site, try to buy the membership-granting subscription – observe it's not allowed

![Captura de tela de 2024-08-07 12-42-47](https://github.com/user-attachments/assets/341a512c-dbd8-429f-acb6-bc64377c2ff9)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207352955560208
